File: Ahorn/entities/starCrystal.jl
module StarCrystal

using ..Ahorn, Maple

@mapdef Entity "vitellary/starcrystal" Crystal(x::Integer, y::Integer, oneUse::Bool=false, time::Number=2.0,
changeDashes::Bool=true, changeInvuln::Bool=true, changeStamina::Bool=true, respawnTime::Number=2.5)

const placements = Ahorn.PlacementDict(
    "Star Crystal (Crystalline)" => Ahorn.EntityPlacement(
        Crystal
    )
)

Ahorn.editingOrder(entity::Crystal) = String["x", "y", "time", "respawnTime", "changeInvuln", "changeDashes", "changeStamina", "oneUse"]

function Ahorn.selection(entity::Crystal)
    x, y = Ahorn.position(entity)

    return Ahorn.getSpriteRectangle("objects/crystals/star/idle00.png", x, y, jx=0.5, jy=0.5)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Crystal) = Ahorn.drawSprite(ctx, "objects/crystals/star/idle00.png", 0, 0, jx=0.5, jy=0.5)

end

File: Loenn/entities/star_crystal.lua
local starCrystal = {}

starCrystal.name = "vitellary/starcrystal"
starCrystal.depth = -100

starCrystal.fieldInformation = {
    time = {
        minimumValue = 0.0
    },
    respawnTime = {
        minimumValue = 0.0
    }
}

starCrystal.placements = {
    {
        name = "star_crystal",
        data = {
            oneUse = false,
            time = 2.0,
            changeDashes = true,
            changeInvuln = true,
            changeStamina = true,
            respawnTime = 2.5
        }
    }
}

starCrystal.texture = "objects/crystals/star/idle00"

return starCrystal
